fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/synchronizedsession/SynchronizedSessionRunner.java

```diff
@@ -42,30 +42,63 @@
 import java.util.function.Function;
 
 /**
- * It delegates all methods to {@link FDBDatabaseRunner}. For {@code run} and {@code runAsync} methods that take the
- * work {@code retriable}, it injects necessary operations to keep the session synchronized by wrapping the work} with
- * {@link SynchronizedSession#workInSession(Function)} or {@link SynchronizedSession#workInSessionAsync(Function)}.
+ * <p>
+ * An {@link FDBDatabaseRunnerInterface} implementation that performs all work in the context of  a
+ * {@link com.apple.foundationdb.record.provider.foundationdb.synchronizedsession.SynchronizedSession}.
+ * </p>
+ * <p>
+ * For all variations of {@code run} and {@code runAsync} methods, the work in the {@code retriable} lambda
+ * is wrapped by calls that check locks and update leases to ensure that two synchronized sessions are not
+ * concurrently running at the same time.
+ * </p>
+ * @see com.apple.foundationdb.record.provider.foundationdb.synchronizedsession.SynchronizedSession
  */
 @API(API.Status.EXPERIMENTAL)
 public class SynchronizedSessionRunner implements FDBDatabaseRunnerInterface {
 
     private FDBDatabaseRunner underlying;
     private SynchronizedSession session;
 
-    // Start a new session. A synchronized session keeps its lock by updating the timestamp in each of its working
-    // transactions, so do NOT try to get the synchronized runner until it is going to be actively used.
+
+    /**
+     * <p>
+     * Produces a new runner, wrapping a given runner, which performs all work in the context of a new
+     *      * {@link com.apple.foundationdb.record.provider.foundationdb.synchronizedsession.SynchronizedSession}.
+     * </p>
+     * <p>
+     * The returned runner will have acquired and started the lease, so care must be taken to ensure that
+     * work begins before the lease expiration period.
+     * </p>
+     * <p>
+     * This is a blocking call.
+     * </p>
+     * @see FDBDatabaseRunner#startSynchronizedSession(Subspace, long)
```

Comment:
I think we generally put `@see` clauses as the last tag in a Javadoc comment. I think there are a few other places like that in this PR.